lib/exercises.rb

```diff
@@ -1,21 +1,56 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  anagrams_hash = Hash.new
+    
+  strings.each do |string|
+    alphabetically_sorted_string = string.chars.sort
+      if anagrams_hash[alphabetically_sorted_string]
+        anagrams_hash[alphabetically_sorted_string] << string
+      else
+        anagrams_hash[alphabetically_sorted_string]=[string]
+      end
+  end
+
+  output_array = []
+  anagrams_hash.each do |key,value|
+    output_array << value
+  end
+
+  return output_array
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) because of the .sort_by used mid method
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
👍 Because you're sorting the (and not in a loop) your algorithm is O(n log n) because Ruby uses MergeSort internally.